controllers/nutanixmachine_controller.go

```diff
@@ -752,7 +859,7 @@ func (r *NutanixMachineReconciler) addGuestCustomizationToVM(rctx *nctx.MachineC
 	return nil
 }
 
-func getDiskList(rctx *nctx.MachineContext) ([]*prismclientv3.VMDisk, error) {
+func getDiskList(rctx *nctx.MachineContext, peUUID string) ([]*prismclientv3.VMDisk, error) {
```

Comment:
nit: Since the PE UUID can be derived or directly read from rctx.NutanixMachine.Spec.Cluster, I think peUUID is redundant. However, we can refactor this in a future PR.

Reply:
Cyclomatic complexity will become higher if we'll get PE uuid inside the `getDiskList`. We already have `peUUID` in context where we are calling the function.
And also in future we can meet requirement to create multi-PE cluster.

So I think we can decide if we would like to get rid of `peUUID` in future.